predicate/lint/linter.py

```diff
@@ -0,0 +1,98 @@
+"""
+Copyright 2022 Gravitational, Inc
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+
+import traceback
+from pathlib import Path
+import yaml
+from lint.parser import get_policy, get_rules
+from lint.rule import NoAllow
+from lint.report import Report, ErrorReport
+from lint.constants import RuleCategory
+
+
+class Linter:
+    """Linter tests given file or directory agains various linter rules"""
+
+    def __init__(self, policy_file_path: str):
+        self.policy_file_path = policy_file_path
+        self.config = self.get_lint_config()
+        # Reports and error occured during lint runner should be collected here
+        self.reports = []
+        self.lint_errors = []
+        self.policies_filepath = []
+
+        self.collect_policy_files(policy_file_path)
+
+    def collect_policy_files(self, policy_file_path):
+        """Collect all policy files if directory is given"""
+        file_or_dir = Path(policy_file_path)
+        if file_or_dir.exists() is False:
+            raise FileNotFoundError
+        if file_or_dir.is_dir():
+            for file in Path(file_or_dir).glob("*.py"):
+                # collect policy files except the ones that trails with __
+                # filters files or directories such as __init__, __pycache__ etc.
+                if "__" not in file.stem:
+                    self.policies_filepath.append(str(file))
+        else:
+            self.policies_filepath.append(policy_file_path)
+
+    def get_lint_config(self):
+        """Returns linter config file"""
+        # TODO: remove hardcoded config file location once desired config location path is decided.
```

Comment:
@klizhentas thoughts on this? iirc most linters for modern ecosystems like Rust/JS/TS do this by starting at the current working dir, searching for this file. If not found, go to the parent, rinse and repeat until you hit a permission error or the root.

Reply:
I created a ticket to track this - https://github.com/gravitational/predicate-lang/issues/79 and will resolve in another PR.